ecc_enclave/enclave/shim.cpp

```diff
@@ -103,17 +109,49 @@ void get_public_state(
 
     LOG_DEBUG("Enclave: got state for key=%s len=%d val='%s'", key, *val_len,
         (*val_len > 0 ? (std::string((const char*)val, *val_len)).c_str() : ""));
+
+    // save in rw set: only save the first key/value-hash pair
+    std::string key_s(key);
+    ByteArray value_hash = pdo::crypto::ComputeMessageHash(ByteArray(val, val + *val_len));
+    auto it = ctx->read_set.find(key_s);
+    if (it == ctx->read_set.end())
+    {
+        // key not found, insert it
+        ctx->read_set.insert({std::string(key), value_hash});
+    }
+    else
+    {
+        // key found, ensure value is the same, or fail
```

Comment:
We should mention here a comment on our assumptions/experiments on (surprising) fabric behavior and maybe a TODO to double-check with fabric maintainers ...  (i.e., along the lines of the text you have in the PR itself)

Note that auction_test actually fails for two tests which might be related to this (or below write assumption)?

Reply:
> We should mention here a comment on our assumptions/experiments on (surprising) fabric behavior and maybe a TODO to double-check with fabric maintainers ... (i.e., along the lines of the text you have in the PR itself)

added a comment

> Note that auction_test actually fails for two tests which might be related to this (or below write assumption)?

The problem was the handling of the composite keys in the FPC go shim :(

Reply:
added comment